worker-courses/providers/udacity-fetcher.ts

```diff
@@ -0,0 +1,22 @@
+import { CourseFetcher } from './fetcher-interface';
+import { Course } from '../models/course';
+
+import * as WebRequest from 'web-request';
+
+export class UdacityFetcher implements CourseFetcher {
+
+  async getAllCourses(): Promise<Course[]> {
+    let url = 'https://www.udacity.com/public-api/v0/courses';
+    let courseData = await WebRequest.json<any>(url);
+    courseData = courseData.courses;
+    let courses = [];
+    for (let course of courseData) {
+      let courseObject = new Course(course.title, course.subtitle, course.homepage)
+      courses.push(courseObject);
+    }
+    return new Promise<Course[]>((resolve, reject) => {
```

Comment:
@askmon @felipegoncalvesmarques  `async` keyword automatically wraps the returning value of a function into a Promise. So instead of using `promise.resolve` you can simply return the value:

```typescript
async myFunc(): Promise<Something> {
  if (everythingIsOk) {
    return new Something(); // with async its the same as Promise.resolve()
  } else {
    throw 'Buuuuu!!!!'; // with async, its the same as Promise.reject()
  }
}
```